.github/scripts/generate_ci_workflows.py

```diff
@@ -1,123 +1,107 @@
-        "test_runner_type": test_runner_type,
-        "cuda_version": cuda_version,
-        "on_pull_request": on_pull_request,
-        "only_build_on_pull_request": only_build_on_pull_request and on_pull_request,
-        "is_scheduled": is_scheduled,
-        "num_test_shards": num_test_shards,
-    }
 
+@dataclass
+class CIWorkflow:
+    # Required fields
+    arch: str
+    build_environment: str
+    test_runner_type: str
 
-LINUX_CPU_TEST_RUNNER = "linux.2xlarge"
-LINUX_CUDA_TEST_RUNNER = "linux.8xlarge.nvidia.gpu"
+    # Optional fields
+    cuda_version: str = ''
+    docker_image_base: str = ''
+    enable_doc_jobs: bool = False
+    exclude_test: bool = False
+    is_libtorch: bool = False
+    is_scheduled: str = ''
+    num_test_shards: int = 1
+    on_pull_request: bool = False
+    only_build_on_pull_request: bool = False
+
+    # The following variables will be set as environment variables,
+    # so it's easier for both shell and Python scripts to consume it if false is represented as the empty string.
+    enable_jit_legacy_test: YamlShellBool = "''"
+    enable_multigpu_test: YamlShellBool = "''"
+    enable_nogpu_no_avx_test: YamlShellBool = "''"
+    enable_nogpu_no_avx2_test: YamlShellBool = "''"
+    enable_slow_test: YamlShellBool = "''"
 
+    def __post_init__(self) -> None:
+        if self.is_libtorch:
+            self.exclude_test = True
+        self.only_build_on_pull_request = self.only_build_on_pull_request and self.on_pull_request
 
-def PyTorchLinuxWorkflow(
-    *,
-    build_environment: str,
-    docker_image_base: str,
-    test_runner_type: str,
-    on_pull_request: bool = False,
-    enable_doc_jobs: bool = False,
-    enable_jit_legacy_test: YamlShellBool = "''",
-    enable_multigpu_test: YamlShellBool = "''",
-    enable_nogpu_no_avx_test: YamlShellBool = "''",
-    enable_nogpu_no_avx2_test: YamlShellBool = "''",
-    enable_slow_test: YamlShellBool = "''",
-    num_test_shards: int = 1,
-    is_scheduled: Optional[str] = None,
-    is_libtorch: bool = False,
-    exclude_test: bool = False,
-) -> PyTorchWorkflow:
-    return {
-        "build_environment": build_environment,
-        "docker_image_base": docker_image_base,
-        "test_runner_type": test_runner_type,
-        "on_pull_request": on_pull_request,
-        "is_scheduled": is_scheduled,
-        "enable_doc_jobs": enable_doc_jobs,
-        "enable_jit_legacy_test": enable_jit_legacy_test,
-        "enable_multigpu_test": enable_multigpu_test,
-        "enable_nogpu_no_avx_test": enable_nogpu_no_avx_test,
-        "enable_nogpu_no_avx2_test": enable_nogpu_no_avx2_test,
-        "enable_slow_test": enable_slow_test,
-        "num_test_shards": num_test_shards,
-        "is_libtorch": is_libtorch,
-        "exclude_test": is_libtorch or exclude_test,   # libtorch is build only
-    }
+    def assert_valid(self) -> None:
+        assert self.arch in ['linux', 'windows'], f"invalid arch: {self.arch}, must be one of ['linux','windows']"
```

Comment:
rather than doing this with a dynamic check, why not just make the type of `arch` be `Literal["linux", "windows"]` instead of just `str`?